action.yml

```diff
@@ -77,10 +77,146 @@ runs:
           }
         }
 
+        function Invoke-Download {
```

Comment:
Consider extracting to a Powershell file and dot-sourcing it `. $PSScriptRoot\Utilities.ps1`

Reply:
That is tempting, but outside the scope of this change.  I think that getting this unblocked first is the important thing.